addons/account/models/account_move.py

```diff
@@ -1,349 +1,1869 @@
+        - The default journal is determined by the default type found in the context or 'entry'.
+        '''
+        move_type = self._context.get('default_type', 'entry')
+        journal_type = 'general'
+        company_id = self._context.get('default_company_id', self.env.company.id)
+        if move_type in ('out_invoice', 'out_refund', 'out_receipt'):
+            journal_type = 'sale'
+        elif move_type in ('in_invoice', 'in_refund', 'in_receipt'):
+            journal_type = 'purchase'
+
+        if self._context.get('default_journal_id'):
+            journal = self.env['account.journal'].browse(self._context['default_journal_id'])
+
+            if move_type != 'entry' and journal.type != journal_type:
+                raise UserError(_("Cannot create an invoice of type %s with a journal having %s as type.") % (move_type, journal_type))
+        else:
+            domain = [('company_id', '=', company_id), ('type', '=', journal_type)]
+            journal = self.env['account.journal'].search(domain, limit=1)
+
+            if not journal:
+                raise UserError(_('Please define an accounting %s journal in your company.') % journal_type)
+        return journal
+
+    @api.model
+    def _get_default_currency(self):
+        ''' Get the default currency from either the journal or either the company. '''
+        journal = self._get_default_journal()
+        return journal.currency_id or journal.company_id.currency_id
+
+    @api.model
+    def _get_default_invoice_incoterm(self):
+        ''' Get the default incoterm for invoice. '''
+        return self.env.company.incoterm_id
+
+    # ==== Business fields ====
+    name = fields.Char(string='Number', required=True, readonly=True, copy=False, default='/')
+    date = fields.Date(string='Date', required=True, index=True, readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=fields.Date.context_today)
+    ref = fields.Char(string='Reference', copy=False, readonly=True,
+        states={'draft': [('readonly', False)]})
+    narration = fields.Text(string='Internal Note')
+    state = fields.Selection(selection=[
+            ('draft', 'Unposted'),
+            ('posted', 'Posted'),
+            ('cancel', 'cancelled')
+        ], string='Status', required=True, readonly=True, copy=False, tracking=True,
+        default='draft')
+    type = fields.Selection(selection=[
+            ('entry', 'Journal Entry'),
+            ('out_invoice', 'Customer Invoice'),
+            ('out_refund', 'Customer Credit Note'),
+            ('in_invoice', 'Vendor Bill'),
+            ('in_refund', 'Vendor Credit Note'),
+            ('out_receipt', 'Sales Receipt'),
+            ('in_receipt', 'Purchase Receipt'),
+        ], String='Type', required=True, store=True, index=True, readonly=True, tracking=True,
+        default="entry")
+    user_id = fields.Many2one('res.users', readonly=True, copy=False, tracking=True,
```

Comment:
rename into invoice_user_id

Reply:
I didn't make it as `user_id` is to painful to be grepped.

Reply:
Come on, only 3900 occurrences in 801 files, each to be checked one by one.

I wrote a small script, I think it should work:
```python
'''Quick grep implementation with a condition on the content of the file'''
import re
import sys
import glob

search_if = sys.argv[1]
search_term = sys.argv[2]
path = sys.argv[3]
file_list = glob.glob(path, recursive=True)

for f in file_list:
    found = False
    file_lines = [l for l in open(f, 'r')]
    joined = "".join(file_lines)
    if not (re.search(search_if, joined) and re.search(search_term, joined)):
        continue
    print(f)
    for i, fr in enumerate(file_lines):
        if re.search(search_term, fr):
            print(str(i) + ':   ' + fr)
```

then I run it with `python3 grepif.py account.move user_id "/path_to_repos/**/*.py"`
There are lot less things to check this way...